bouncer/src/repo/core/model/bson/repo_bson_factory.h

```diff
@@ -263,33 +277,42 @@ namespace repo {
 				* @return returns a mesh node
 				*/
 				static MeshNode makeMeshNode(
-					const std::vector<repo::lib::RepoVector3D>        &vertices,
-					const std::vector<repo_face_t>                    &faces,
-					const std::vector<repo::lib::RepoVector3D>        &normals,
-					const std::vector<std::vector<float>>             &boundingBox,
-					const std::vector<std::vector<repo::lib::RepoVector2D>>   &uvChannels,
-					const std::vector<repo_color4d_t>                 &colors,
-					const std::string                                 &name,
-					const std::vector<repo::lib::RepoUUID>            &parents,
-					const int                                         &apiLevel = REPO_NODE_API_LEVEL_1);
+					const std::vector<repo::lib::RepoVector3D>& vertices,
+					const std::vector<repo_face_t>& faces,
+					const std::vector<repo::lib::RepoVector3D>& normals,
+					const std::vector<std::vector<float>>& boundingBox,
+					const std::vector<std::vector<repo::lib::RepoVector2D>>& uvChannels,
+					const std::string& name,
+					const std::vector<repo::lib::RepoUUID>& parents);
 
 				static MeshNode makeMeshNode(
 					const std::vector<repo::lib::RepoVector3D>& vertices,
 					const std::vector<repo_face_t>& faces,
 					const std::vector<repo::lib::RepoVector3D>& normals,
 					const std::vector<std::vector<float>>& boundingBox,
-					const std::vector<std::vector<repo::lib::RepoVector2D>>& uvChannels,
-					const std::vector<repo_color4d_t>& colors);
+					const std::vector<std::vector<repo::lib::RepoVector2D>>& uvChannels);
+
+				static MeshNode makeMeshNode(
+					const std::vector<repo::lib::RepoVector3D>& vertices,
+					const std::vector<repo_face_t>& faces,
+					const std::vector<repo::lib::RepoVector3D>& normals,
+					const std::vector<std::vector<float>>& boundingBox)
+				{
+					return makeMeshNode(vertices, faces, normals, boundingBox,
```

Comment:
A bit confused why we have 3 different function declarations for makeMeshNodes

Can't we just have 1 with default values declared for optional parameters?

Reply:
Done!